### PR TITLE
lesson03 exercises

### PR DESCRIPTION
**Pull request by bcamp3**
adding lesson03 exercises:
- slicing_lab.py
- list_lab.py
- strformat_lab.py

adding lesson03, Assignment 1:
- mailroom.py

### CODE REVIEW

**Comment by bcamp3** [comment]
Added Assignment 1: Mailroom.   mailroom.py script

File: students/bcamp3/lesson03/list_lab.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+List Lab
+
+"""
+
+"""
+Series 1
+"""
+print('----------\n Series 1\n----------')
+
+fruits = ['Apples','Pears','Oranges','Peaches']
```
Comment (charlesrobison) [inline students/bcamp3/lesson03/list_lab.py:13]: Missing whitespaces after commas

File: students/bcamp3/lesson03/list_lab.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+if response[-1] != 's':
+    response += 's'
+
+fruits.append(response.capitalize())
+
+print(fruits)
+
+response = 0
+while response >5 or response <1:
```
Comment (charlesrobison) [inline students/bcamp3/lesson03/list_lab.py:27]: Missing whitespace after operators

File: students/bcamp3/lesson03/list_lab.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    response = int(input(f'\nPick a number between 1 and {len(fruits)} > '))
+
+print(f'\nFruits list item {response:d} is {fruits[response-1]}')
+
+print('\nAdding Bananas to beginning of fruits list:')
+fruits = ['Bananas'] + fruits
+print(fruits)
+
+print('\nAdding Melons to beginning of fruits list:')
+fruits.insert(0,'Melons')
```
Comment (charlesrobison) [inline students/bcamp3/lesson03/list_lab.py:37]: Need whitespace here too

File: students/bcamp3/lesson03/list_lab.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+print(fruits)
+
+print("\nFruits starting with the letter 'P' :")
+for fruit in fruits:
+    if fruit[0]=='P':
```
Comment (charlesrobison) [inline students/bcamp3/lesson03/list_lab.py:42]: Whitespaces around operator here

File: students/bcamp3/lesson03/list_lab.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+if response[-1] != 's':
+    response += 's'
+
+fruits.append(response.capitalize())
+
+print(fruits)
+
+response = 0
+while response >5 or response <1:
+    response = int(input(f'\nPick a number between 1 and {len(fruits)} > '))
```
Comment (charlesrobison) [inline students/bcamp3/lesson03/list_lab.py:28]: Good!

File: students/bcamp3/lesson03/list_lab.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+fruits = ['Apples','Pears','Oranges','Peaches']
+
+print('\nThe fruits list: \n', fruits)
+
+response = input('\nWhat is another fruit that you like? > ')
+
+if response[-1] != 's':
+    response += 's'
+
+fruits.append(response.capitalize())
```
Comment (charlesrobison) [inline students/bcamp3/lesson03/list_lab.py:22]: Nice use of capitalize()!

File: students/bcamp3/lesson03/list_lab.py [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+print('\nThe fruits list: \n', fruits)
+fruits_copy = fruits[:]
+print()
+for fruit in fruits_copy:
+    response = input(f'Do you like {fruit:<8} ? > ')
+    while response.lower() not in ['yes','no']:
+        print(f"*** Please respond with 'Yes' or 'No' ***")
+        response = input(f'Do you like {fruit:<8} ? > ')
+    if response.lower() == 'no':
+        fruits.remove(fruit)
```
Comment (charlesrobison) [inline students/bcamp3/lesson03/list_lab.py:91]: Works well!

File: students/bcamp3/lesson03/mailroom.py [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+This script accepts user input to perform the following donation database tasks:
+    1 - Display the donor database list or update the donation database with a new donation.  
+        Then display a 'Thank You' message to the donor of the new donation.
+    2 - Display a donation database report
+    3 - Quit out of the script
+"""
+
+#creating the donor database using nested lists:
+donors = [['Katherine Elmhurst','David Anderson','Edward Hoffstein','Rebecca Manriquez','Callum Foxbright'],
+          [[1000.,1500.,1900.]   , [10865.,5750.]     , [200.,200.,200.]    , [1750.,1500.]         , [130.75]]]
```
Comment (charlesrobison) [inline students/bcamp3/lesson03/mailroom.py:15]: Missing whitespaces after commas

Comment (charlesrobison) [reply students/bcamp3/lesson03/mailroom.py:15]: Here as well as other areas below in this code

File: students/bcamp3/lesson03/mailroom.py [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+    while opt not in ['1','2','3']:
+        opt = input(" Please choose option '1', '2', or '3' > ")
+
+    if opt == '1':
+        send_thank_you()
+    elif opt == '2':
+        create_report()
+    elif opt == '3':
+        print('Quitting...')
```
Comment (charlesrobison) [inline students/bcamp3/lesson03/mailroom.py:81]: You'll want the program to continue until the user selects 'Quit'

File: students/bcamp3/lesson03/slicing_lab.py [hunk 10]
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+
+"""
+Slicing Lab
+
+"""
+
+def exchange_first_last(seq):
+    if len(seq)<=1:
```
Comment (charlesrobison) [inline students/bcamp3/lesson03/slicing_lab.py:9]: Missing whitespace around operator

File: students/bcamp3/lesson03/slicing_lab.py [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+    return seq[::2]
+    
+def remove_first4_last4_every_other(seq):
+    return seq[4:-4:2]
+
+def reverse_seq(seq):
+    return seq[::-1]
+
+def thirds_jumble(seq):
+    if len(seq)<=1:
```
Comment (charlesrobison) [inline students/bcamp3/lesson03/slicing_lab.py:24]: Here too

File: students/bcamp3/lesson03/slicing_lab.py [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+        return seq
+    elif len(seq)==2:
```
Comment (charlesrobison) [inline students/bcamp3/lesson03/slicing_lab.py:26]: Here as well

**Comment by bcamp3** [comment]
Thanks Charles,

I took the initial assignment direction too literally when it said that
progress did not have to be saved, and I just missed the sorted report
requirement.  For my own education and practice I have updated the
mailroom.py script to fulfill all of the requirements.

Also, I installed pylint but I am still trying to figure out how to
integrate it into the editor I am using, Notepad++.

-Brian

**Comment by natasha-aleksandrova** [comment]
@bcamp3 I highly recommend Pycharm for code editor, it has a linter built in.

**Pull request merged**